src/app/lab/lab.component.ts

```diff
@@ -23,43 +23,47 @@ import * as _ from 'lodash';
 })
 export class LabComponent implements OnInit, OnDestroy {
 
-  private subscription: Subscription;
-  public project: Project;
+    private subscription: Subscription;
+    public project: Project;
 
-  constructor(private projectService: ProjectService,
-              private route: ActivatedRoute,
-              private router: Router) {
-  }
+    constructor(private projectService: ProjectService,
+                private route: ActivatedRoute,
+                private router: Router) {
+    }
 
-  ngOnInit() {
-    // Load existing project id or redirect to dashboard
-    this.subscription = this.route.params.subscribe(params => {
-      let id: string = params['id'];
-      if (id !== undefined) {
-        this.project = this.projectService.get(id);
-      } else {
-        this.router.navigate(['']);
-      }
-    });
-  }
+    ngOnInit() {
+        // Load existing project id or redirect to dashboard
+        this.subscription = this.route.params.subscribe(params => {
+            let id: string = params['id'];
+            if (id !== undefined) {
+              this.project = this.projectService.get(id);
+            }
+        });
+        // Reroute to dashboard if project doesn't exist
+        if (!this.project) {
```

Comment:
I don't think this is doing what you expect. Shouldn't this check be inside the subscribe handler right after the project is pulled from the ProjectService?

Also consider moving all of the logic in this subscribe onNext handler to a separate named function as a stylistic improvement.